### ISSUE TITLE
docker pull hangs downloading image

### ISSUE THREAD

**Issue by sebastianperruolo** [docker/for-mac#2083]
I tried several times to pull mssql-server-linux but I got always the same result.

```
$ docker pull microsoft/mssql-server-linux
latest: Pulling from microsoft/mssql-server-linux
aed15891ba52: Already exists 
773ae8583d14: Already exists 
d1d48771f782: Already exists 
cd3d6cd6c0cf: Already exists 
8ff6f8a9120c: Already exists 
1fd7e8b10447: Already exists 
bd485157db89: Already exists 
273a1970ce9c: Already exists 
c6f0182f94e9: Downloading [=================================================> ]  275.2MB/275.7MB
47c608bf53ae: Waiting 
```

### Expected behavior

Finish dowloading of mssql-server-linux image

### Actual behavior

It always hang with the same fs file with id c6f0182f94e9

### Information

> Docker for Mac: version: 17.06.2-ce-mac27 (4cdec4294a50b2233146b09469b49937dabdebdd)
> macOS: version 10.12.6 (build: 16G29)
> logs: /tmp/5542512B-8253-4454-9064-2CD6AD31D520/20170922-191740.tar.gz
> [OK]     db.git
> [OK]     vmnetd
> [OK]     dns
> [OK]     driver.amd64-linux
> [OK]     virtualization VT-X
> [OK]     app
> [OK]     moby
> [OK]     system
> [OK]     moby-syslog
> [OK]     db
> [OK]     env
> [OK]     virtualization kern.hv_support
> [OK]     slirp
> [OK]     osxfs
> [OK]     moby-console
> [OK]     logs
> [OK]     docker-cli
> [OK]     menubar
> [OK]     disk

### Steps to reproduce the behavior

  1. Run `docker pull microsoft/mssql-server-linux`
  2. Wait

**Comment by jasonbivins** [comment]
Hi @sebastianperruolo 
Let's see if you can remove what's there and pull down a fresh image - try this command
``` docker image rm -f microsoft/mssql-server-linux```

**Comment by sebastianperruolo** [comment]
Hi @jasonbivins 

I tried that command but I'm afraid that it didn't help:
```
$ docker image rm -f microsoft/mssql-server-linux
Error response from daemon: No such image: microsoft/mssql-server-linux:latest

$ docker pull microsoft/mssql-server-linux
Using default tag: latest
latest: Pulling from microsoft/mssql-server-linux
aed15891ba52: Already exists 
773ae8583d14: Already exists 
d1d48771f782: Already exists 
cd3d6cd6c0cf: Already exists 
8ff6f8a9120c: Already exists 
1fd7e8b10447: Already exists 
bd485157db89: Already exists 
273a1970ce9c: Already exists 
c6f0182f94e9: Downloading [=================================================> ]  275.2MB/275.7MB
47c608bf53ae: Waiting 
```

**Comment by sebastianperruolo** [comment]
Today I tried again and it was able to finish the download. Although I'm not sure what fixed the issue, I'll close it.

**Comment by hervinhio** [comment]
For some reason your issue did not appear at that moment. But I don't think that it means that the issue dissapeard because I'm facing the same issue.

**Comment by arefaslani** [comment]
I'm facing the issue today.
```bash
❯ docker pull jwilder/nginx-proxy
Using default tag: latest
latest: Pulling from jwilder/nginx-proxy
2a72cbf407d6: Already exists
04b2d3302d48: Already exists
e7f619103861: Already exists
668b13d991c2: Downloading [=================>                           ]   1.24MB/3.468MB
9b4e089f640c: Download complete
b055f74a1218: Downloading [=========================>                   ]   1.49MB/2.894MB
1f16e62a7a7f: Downloading [========>                                    ]  469.6kB/2.894MB
73a091d7ccc4: Waiting
eb383481e5a9: Waiting
32a4c9e45ee0: Waiting
```

**Comment by sebastianperruolo** [comment]
@arefaslani @hervinhio I’m reopening the issue

**Comment by YRM64** [comment]
This issue has been tagged over the last several years, as far back as April 2015.
Pulling images sometimes gets stuck, #813, June 2017.
Pull gets stuck waiting for a layer, #22598, May 2016.
Docker can hang indefinitely waiting for a nonexistent process to download the repository, #12823,
   April, 2015.
Docker hangs forever during downloading image layers, Stack Overflow (stackoverflow.com).
Docker Compose stuck downloading or pulling fs layer, Stack Overflow (stackoverflow.com).
Boot2docker - Docker hangs forever during pulling image layers, Stack Overflow (stackoverflow.com).
Docker is stuck on download complete for pull, Stack Overflow (stackoverfow.com) 
Docker stuck on pull - Docker for Mac - Docker Forums (forums.docker.com)
Pulling Docker image is too slow, #1767

**Comment by ent1c3d** [comment]
Is here any fix for this issue?  I still have same problem, I am trying to pull image with command and also with kitematic, but I hangs in both cases..

**Comment by colinPlusPlus** [comment]
I'm also having the same issue with microsoft/mssql-server-linux.

**Comment by MAliNaqvi** [comment]
Another update, for me it seems to be an ISP specific issue. I was able to verify it by switching to a mobile network and pull the python image without it getting stuck.

Wondering if anyone else can replicate this issue; that by switching to another network, the pull doesn't hang.

**Comment by adyz** [comment]
I have the same problem. I think that a timeout or some kind of error would be nice instead if hanging like this...

**Comment by diversemix** [comment]
I have recently experienced the same issue from inside a VM. If it helps, I captured using `tcpdump` and there seems to be endless ACK'ing going on with no data being sent...

```
12:49:09.945089 IP 10.0.2.15.49554 > 104.18.122.25.https: Flags [.], ack 1, win 53960, length 0
12:49:09.945922 IP 104.18.122.25.https > 10.0.2.15.49554: Flags [.], ack 1, win 65535, length 0
12:49:15.199384 IP 10.0.2.15.49570 > 104.18.122.25.https: Flags [.], ack 1, win 53200, length 0
12:49:15.230575 IP 104.18.122.25.https > 10.0.2.15.49570: Flags [.], ack 1, win 65535, length 0
12:49:40.024831 IP 10.0.2.15.49554 > 104.18.122.25.https: Flags [.], ack 1, win 53960, length 0
12:49:40.025511 IP 104.18.122.25.https > 10.0.2.15.49554: Flags [.], ack 1, win 65535, length 0
```

**Comment by zbindenren** [comment]
Same issue here. A restart of docker daemon is not possible. Only solution is to reboot the node.

**Comment by franlab** [comment]
Same issue

**Comment by alok87** [comment]
Pulling images from local works but does not work from machines in AWS regions

**Comment by alok87** [comment]
For us restarting the mirror registry fixed the issue

**Comment by Logician724** [comment]
same issue here, can't seem to pull the node 10 image.

**Comment by picturedots** [comment]
Restarting docker fixed the issue for me.

**Comment by tmwatchanan** [comment]
Same issue here. Even restarting docker service cannot solve it.

**Comment by Pezhvak** [comment]
i'm having constant problem with it, i can't pull mysql:5.7, tried more than 8 times now

**Comment by MAliNaqvi** [comment]
Not a surgical solution, but after running 

```
brew update 
brew upgrade
```

Docker is able to pull images correctly. No problems so far, a month later.

**Comment by jayanderson** [comment]
I ran into this today again (upgraded (`Version 18.06.1-ce-mac73 (26764)`) and restarted to make sure things were clean on this end). Once it hangs the only option is to kill it. It doesn't seem to timeout. Things that would be nice:
- It times out automatically instead of hanging in this state (hopefully with the ability to change the timeout)
- The things previously downloaded aren't totally lost so it can pick up where it left off, avoid duplicating what was previously downloaded, and possibly give the user a path to complete the full download.

**Comment by gururajdori** [comment]
Restarting docker engine helped me

**Comment by mithunj43** [comment]
Unable to find image 'jboss/wildfly:latest' locally
latest: Pulling from jboss/wildfly
aeb7866da422: Downloading [=============>                                     ]  20.35MB/74.7MB
157601a0b538: Download complete 
642f4164f381: Download complete 
ee2abcb9139a: Downloading [============>                                      ]  17.17MB/71.5MB
49bbff283f2b: Downloading [==>                                                ]  9.112MB/178.2MB

Every time struck at the same thing.

**Comment by skbly7** [comment]
I faced the same in `Version 17.03.1-ce-rc1-mac3 (15924)` today and was able to pull image only after docker restart.

```
❯ docker pull nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04
9.0-cudnn7-runtime-ubuntu16.04: Pulling from nvidia/cuda
18d680d61657: Already exists
0addb6fece63: Already exists
78e58219b215: Already exists
eb6959a66df2: Already exists
6ef1ff668c93: Already exists
f5f8f0544aa2: Already exists
3d28d96eb352: Already exists
1b48d63763c4: Downloading 476.6 MB/477.1 MB
467c984074f4: Download complete
^C
❯ docker rm nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04
Error response from daemon: No such container: nvidia/cuda:9.0-cudnn7-runtime-ubuntu16.04
```

**Comment by macx78** [comment]
the simple 'docker pull java' is hanging for me...a bit irritating. any resolution or temp work around yet... ohe yeah, restarting docker did not work for me.

**Comment by gvdm90** [comment]
Same issue with the `postgres` image for me

**Comment by jojoblaze** [comment]
save problem here. 
I'm running docker on a raspberry pi 3. I spotted this problem recently. Some days ago it was working normally.

**Comment by myc0l** [comment]
Same issue with me as well even tried reinstalling but even after I still can't pull mongo it just freezes

**Comment by SchutMJ** [comment]
Same issue with microsoft/dotnet:2.1-runtime image

**Comment by keen4e** [comment]
Same microsoft/dotnet:2.1-runtime :/

**Comment by moluzhui** [comment]
Same issue with the flannel image for me
```
v0.9.1: Pulling from coreos/flannel
6d987f6f4279: Already exists 
16a827ca53c8: Already exists 
8f4dde5859ad: Already exists 
40db9b39f697: Already exists 
48f4941e520f: Already exists 
f77b10f3d93e: Waiting
```

**Comment by flopetegui** [comment]
7.0.1: Pulling from library/logstash
8ba884070f61: Downloading [============>]  63.08MB/63.08MB
2acb8d9c2ee9: Downloading [============>]   94.5MB/94.5MB
e1094286f832: Download complete
335c368bfe0b: Downloading [============>]  164.1MB/164.1MB
7caab92382af: Download complete
92d0a5b69598: Download complete
44fd5e3c8ee5: Download complete
1ea17513e98c: Download complete
6db053fec74e: Download complete
b08c9ba7eb37: Download complete
a1882ed8e8e9: Download complete
read tcp 192.168.65.3:45154->104.18.125.25:443: read: connection reset by peer

This happens after several retires.

**Comment by dholbrook86** [comment]
Reporting same issue.

27833a3ba0a5: Pull complete
394a1d92181c: Pull complete
0efb6deb6da1: Pull complete
81549544b420: Pull complete
e24aea066ae4: Pull complete
4c4cbd20ba91: Pull complete
05bc79342649: Pull complete
0b92851dadd5: Downloading [====>                                              ]  119.4kB/1.269MB
7f1a64f50c08: Download complete
9e31f7e05b9b: Download complete
489bfb1e4741: Download complete
24c26f2acf8f: Download complete
46cbbbfb7296: Download complete
a358f4537703: Download complete
8c7b241589b9: Download complete

**Comment by erssebaggala** [comment]
I am experiencing the same issue - see below.

```
Pulling bts-api (bodastage/bts-ce-api:1.2.1)...
1.2.1: Pulling from bodastage/bts-ce-api
9ff7e2e5f967: Pull complete
59856638ac9f: Pull complete
6f317d6d954b: Pull complete
a9dde5e2a643: Pull complete
1e093bf37a80: Pull complete
8d638c1a19ca: Downloading [======>                                            ]   17.7MB/144.9MB
3b95c41e6689: Download complete
0f2da64ce7f5: Download complete
bb7cf30ae873: Download complete
04bedca1c8ed: Download complete
58eb89d012c4: Downloading [====================>                              ]  57.48MB/140.1MB
169e95234836: Download complete
2a9af58f6055: Download complete
59503fd86137: Download complete
26e065acec44: Download complete
307e84eba6be: Download complete
794e7ffd3e6a: Download complete
```

Below is my system information:
```
$ docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 16
Server Version: 18.05.0-ce
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 160
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.93-boot2docker
Operating System: Boot2Docker 18.05.0-ce (TCL 8.2.1); HEAD : b5d6989 - Thu May 10 16:35:28 UTC 2018
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.858GiB
Name: default
ID: V7P6:2JT3:S3QR:OTOA:UK2H:QG7V:B2ER:OWNO:KV2Y:4T6L:7K4B:OF36
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: bodastage
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

Below is extract from  /var/log/docker.log
```
time="2019-05-23T23:08:44.271712209Z" level=info msg="ignoring event" module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
time="2019-05-23T23:08:53.188484540Z" level=info msg="Container 36b5cd7d20f7bea38ca02e3fa881403297caf47aab4de67d6ac5be074eb924ee failed to exit within 10 seconds of signal 15 - using the force"
time="2019-05-23T23:08:53Z" level=info msg="shim reaped" id=36b5cd7d20f7bea38ca02e3fa881403297caf47aab4de67d6ac5be074eb924ee module="containerd/tasks"
time="2019-05-23T23:08:53.384820450Z" level=info msg="ignoring event" module=libcontainerd namespace=moby topic=/tasks/delete type="*events.TaskDelete"
time="2019-05-24T00:02:51.075292589Z" level=error msg="Not continuing with pull after error: context canceled"
time="2019-05-24T01:35:49.979243384Z" level=error msg="Not continuing with pull after error: context canceled"
time="2019-05-24T09:35:08.549384052Z" level=error msg="Download failed, retrying: read tcp 10.0.2.15:47726->104.18.121.25:443: read: connection reset by peer"
time="2019-05-24T09:35:08.550419178Z" level=error msg="Not continuing with pull after error: context canceled"
time="2019-05-24T09:59:00.782407161Z" level=error msg="Download failed, retrying: unexpected EOF"
time="2019-05-24T12:25:13.952890629Z" level=error msg="Not continuing with pull after error: context canceled"
```

**Comment by andreportela** [comment]
same thing here =( do we have any sort of fallback to try to unblock the pulling?

**Comment by lucasgcb** [comment]
Still happens, particularly nasty with projects which contain several layers.
A timeout and retry would be fantastic.

**Comment by Amenophis1** [comment]
Same issue, with the majority of the pulls I tried. I don't know how to fix it.

**Comment by YamYamM** [comment]
Same issue, restarting docker didn't help, issue presents itself in the majority of the pulls. very annoying.

**Comment by thebmusic** [comment]
Same as everyone else, I'm seeing this with multiple images.  Sometimes it gets stuck in the middle of downloading, sometime when pulling fs layer.  It's very frustrating, like how is this acceptable?

**Comment by MohammedEssehemy** [comment]
same issue, solved after quitting and reopening `Docker Desktop`.

**Comment by mulho** [comment]
Same thing here. Can't download a large docker image from AWS ECR when working from home. The connection gets stalled forever for some layers. If I leave it overnight, it eventually finishes, but with an "unexpected EOF" error. The same image downloads fine when I'm on my office network. Just tried upgrading to the latest "19.03.1, build 74b1e89", restarting, but the connection always gets stalled at some point. This is basically stopping me from working from home :(

**Comment by AlexZhovnuvaty** [comment]
"service docker restart" solved the issue for me

**Comment by mulho** [comment]
So, in my case, it looks like my router had some high package loss. I changed the router and it now works. But, normal browsing would work fine. So it still lools like docker fetch resilience has some space for improvements. Auto restarts or something like that.

**Comment by dogukanturan** [comment]
`service restart docker` did not solve my problem.

**Comment by rodrigogs** [comment]
Same here

**Comment by drtwisted** [comment]
It's already been two years. And still I keep facing the issue over and over again. The latest is just now.

**Comment by MGD1981** [comment]
I see that there is a Docker For Windows issue open for this as well (https://github.com/docker/for-win/issues/813) -- does anyone know if there is one for Linux Docker? Because we're encountering this bug in our Swarm running on CentOS 7 VMs frequently and it's halting productivity.

**Comment by da3monHunter** [comment]
> I see that there is a Docker For Windows issue open for this as well ([docker/for-win#813](https://github.com/docker/for-win/issues/813)) -- does anyone know if there is one for Linux Docker? Because we're encountering this bug in our Swarm running on CentOS 7 VMs frequently and it's halting productivity.

Faced this on centos7 too

**Comment by paullaffitte** [comment]
I have this issue too with mariadb:10.4 on ubuntu.

**Comment by erasmas** [comment]
In my case rebooting my macbook helped. 🙈

**Comment by evan-wehi** [comment]
Also having the issue in an AWS ubuntu instance. Rebooting doesn't help.

**Comment by andriihomiak** [comment]
Had same issue with `puckel/docker-airflow` image on ubuntu. 
```$ sudo systemctl stop docker.service && sudo systemctl start docker.service```
did the trick for me

**Comment by katopz** [comment]
![image](https://user-images.githubusercontent.com/97060/68321894-11922780-00f5-11ea-90e4-37f1d3ca11f6.png)
```bash
katopz@kat ~ % docker pull microsoft/mssql-server-linux
Using default tag: latest
latest: Pulling from microsoft/mssql-server-linux
59ab41dd721a: Downloading  39.85MB/42.22MB
57da90bec92c: Download complete 
06fe57530625: Download complete 
5a6315cba1ff: Download complete 
739f58768b3f: Download complete 
0b751601bca3: Download complete 
bcf04a22644a: Downloading  59.24MB/62.29MB
6b5009e4f470: Downloading  278.4MB/281.1MB
a9dca2f6722a: Waiting 
```
Restart didn't do the trick, I still waiting here...

**Comment by steklopod** [comment]
Same issue:

```sh
PS C:\WINDOWS\system32> docker image rm -f node:latest

PS C:\WINDOWS\system32> docker pull node:latest

latest: Pulling from library/node
9a0b0ce99936: Downloading [============================>  ]  44.23MB/45.38MB        
```

`service restart docker` gives me:
```sh
Get-Service : Не удается найти позиционный параметр, принимающий аргумент "docker".
строка:1 знак:1
+ service restart docker
+ ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Get-Service], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.GetServiceCommand
```

**Comment by mokalovesoulmate** [comment]
I am a new Docker user and would trying to use [Devilbox](http://devilbox.org/). I have the same issue, docker-compose seem to download a lot of files but some of them were stalled, so I aborted it by pressing Ctrl+C.

But the most annoying issue is each time I cancel and resume it, it keeps redownload the files that already downloaded before I cancel. (This is Docker issue, I believe)

I was stuck for an hour and did not have other network option. I believe the issue is either my own ISP or the server where the files were hosted.

**Comment by MahmoudSoliman922** [comment]
I had the same issue and changing the "max-concurrent-downloads" to 1 solved it for me.

**Comment by Thirupatel** [comment]
I am getting below error... while pulling slenium git hub
C:\Users\jrohilla>docker pull selenium/hub
Using default tag: latest
latest: Pulling from selenium/hub
7ddbc47eeb70: Already exists
c1bbdc448b72: Already exists
8c3b70e39044: Already exists
45d437916d57: Already exists
224ff5c95ca1: Already exists
acaa6aac1124: Already exists
0b8f55da2e47: Already exists
5670ee04a00a: Already exists
ebf77b0e3974: Already exists
25ba20a3fc45: Already exists
1701de5595aa: Already exists
24acfc5a01c8: Waiting

**Comment by mkorenkov** [comment]
`max-concurrent-downloads` did not solve the issue for me. Tried different networks, Wi-FI and ethernet. Can this be an issue on hub.docker.com side?

downloading golang:1.13.7 is like

attempt 1 (and stuck):
bd20b51e1705: Downloading [===============================>                   ]  75.53MB/120.1MB

attempt 2 (and stuck):
bd20b51e1705: Downloading [=====================================>             ]  88.93MB/120.1MB

attempt 3 (and stuck):
bd20b51e1705: Downloading [=====================================>             ]  91.18MB/120.1MB

attempt 4 (and stuck):
bd20b51e1705: Downloading [=======================================>           ]  94.46MB/120.1MB

**Comment by JnMik** [comment]
Same issue here, my image is getting pulled from aws ECR so I don't think it's hub.docker.com's fault @mkorenkov

**Comment by JnMik** [comment]
What I can add is, i'm pulling the image from a device behind a very slow ISP (xplorenet, satellite). 
My guess is they were throttling me for abuse of network... but maybe not cuz definitely we are many with the same issue.

**Comment by l-qing** [comment]
In my scene, trigger reconnect, by restart network or close hang tcp socket, can finish the download.

I use Surge on MacOS, it's can show the active socket in dashboard.
It can easy break the hang socket to trigger reconnect.

**Comment by manjotsingh99** [comment]
> I tried several times to pull mssql-server-linux but I got always the same result.
> 
> ```
> $ docker pull microsoft/mssql-server-linux
> latest: Pulling from microsoft/mssql-server-linux
> aed15891ba52: Already exists 
> 773ae8583d14: Already exists 
> d1d48771f782: Already exists 
> cd3d6cd6c0cf: Already exists 
> 8ff6f8a9120c: Already exists 
> 1fd7e8b10447: Already exists 
> bd485157db89: Already exists 
> 273a1970ce9c: Already exists 
> c6f0182f94e9: Downloading [=================================================> ]  275.2MB/275.7MB
> 47c608bf53ae: Waiting 
> ```
> 
> ### Expected behavior
> Finish dowloading of mssql-server-linux image
> 
> ### Actual behavior
> It always hang with the same fs file with id c6f0182f94e9
> 
> ### Information
> > Docker for Mac: version: 17.06.2-ce-mac27 (4cdec4294a50b2233146b09469b49937dabdebdd)
> > macOS: version 10.12.6 (build: 16G29)
> > logs: /tmp/5542512B-8253-4454-9064-2CD6AD31D520/20170922-191740.tar.gz
> > [OK]     db.git
> > [OK]     vmnetd
> > [OK]     dns
> > [OK]     driver.amd64-linux
> > [OK]     virtualization VT-X
> > [OK]     app
> > [OK]     moby
> > [OK]     system
> > [OK]     moby-syslog
> > [OK]     db
> > [OK]     env
> > [OK]     virtualization kern.hv_support
> > [OK]     slirp
> > [OK]     osxfs
> > [OK]     moby-console
> > [OK]     logs
> > [OK]     docker-cli
> > [OK]     menubar
> > [OK]     disk
> 
> ### Steps to reproduce the behavior
> 1. Run `docker pull microsoft/mssql-server-linux`
> 2. Wait
THE ISSUE IS RESOLVED.
THE MAIN PROBLEM WAS IN THE TIMEOUT ERROR. SO JUST TO RESOLVE THIS CHANGE THE WIFI CONNECTION WHENEVER U GOT STUCK AND THEN CHANGE IT BACK.WORKS EVERYTIME.KUDOS!!!!!!!!!!!!!!

**Comment by manjotsingh99** [comment]
THE ISSUE IS RESOLVED.
THE MAIN PROBLEM WAS IN THE TIMEOUT ERROR. SO JUST TO RESOLVE THIS CHANGE THE WIFI CONNECTION WHENEVER U GOT STUCK AND THEN CHANGE IT BACK.WORKS EVERYTIME.KUDOS!!!!!!!!!!!!!!

**Comment by tuxayo** [comment]
@manjotsingh99 
> THE ISSUE IS RESOLVED.

Unfortunately this is just a workaround. And it's not very usable on a server.

**Comment by manjotsingh99** [comment]
> @manjotsingh99
> 
> > THE ISSUE IS RESOLVED.
> 
> Unfortunately this is just a workaround. And it's not very usable on a server.

I know but this works and no other solution is out there. what to do now!!

**Comment by meridsa** [comment]
Bump. Facing the same issue.

**Comment by tuxayo** [comment]
@meridsa No need to bump. There are already a lot messages here which makes it hard to follow or to come here for the first time.
The often recommended alternative is to upvote the issue: https://github.com/docker/for-mac/issues/2083#issue-259952621

**Comment by Truth1984** [comment]
Suggestion:  Consider caching already downloaded content for quick restart ?

**Comment by podjackel** [comment]
> I was running into this issue using docker from ubuntu inside VBox. After changing my virtual machine's network configuration from NAT to Bridged Adapter it worked smoothly.
> 
> Hope it helps!

This seems to work in 2020. Given all the issues and different levels of fixes, I'm going to guess this is an issue with how the docker client handles network interruptions or stagnation.

**Comment by michael-bud** [comment]
I had this occur after I interrupted (Ctrl + C) a docker pull in the middle of it downloading the said layer. A restart of docker daemon fixed it.

So based on that, plus what a lot of others have said, it appears to occur when a download is interrupted in some way?

**Comment by ikingye** [comment]
restart docker fix it.

`systemctl restart docker`

**Comment by alexivkin** [comment]
restarting dockerd, reconnecting network, closing hang tcp socket... Why after all these years nobody bothered fixing the darn bug in the docker code itself? Is it too hard to add a timeout/reconnect on a hung connection? :disappointed:

**Comment by francilioaraujo** [comment]
Had the same issue today pulling redis:latest inside a debian VM. `systemctl restart docker` solved it for me. It bothers me that if this happens in production for someone the only way to solve is to restart the daemon

**Comment by N4SIRODDIN3** [comment]
Hi,
I figured out that the download progress happens in the background. When I pressed the enter key, I managed to continue the download progress.
Hope this will help.

**Comment by andranik-muradyan-vineti** [comment]
Same here, on multiple Ubuntu 20.04 machines. The hello-world downloads just fine thou. Restarting the machines and the service did not do the trick for me.

**Comment by DavidJulienMillet** [comment]
Hi i had the same problem. If the Image part has begin its download with a specific connection (my home wifi), it can't be finished with my phone 4G wifi. I solve the solution switching again to my home wifi to finish download even if its way  much longer.
So I think exploring the internet configuration can be a  solution to this problem...

**Comment by richiethom** [comment]
In my experience this is due to non-standard connections. Eg a corporate VPN over a slow domestic wifi-->ADSL connection. Downloading the same image on the same machine, but on the fast corporate network, finished absolutely fine. Downloading the same image on my own computer (rather than employer's) without their VPN software, no problem either.

**Comment by larytet** [comment]
In my case this is a VPN+WiFi related issue. It would be nice to have more debug information when the download fails.

**Comment by Remie9** [comment]
Rebooting helped for me

**Comment by crypt3lx2k** [comment]
I had this for a couple of months where big downloads would not finish in docker. All issues were resolved when I switched out my router.

**Comment by aravinthkumar** [comment]
ok I see this is open for 3 years, I am not sure if this would be addressed any day soon but even I have the same issue today :(

Stuck

7ddbc47eeb70: Pull complete                                                                                                                                                                                                                  c1bbdc448b72: Pull complete                                                                                                                                                                                                                  8c3b70e39044: Pull complete                                                                                                                                                                                                                  45d437916d57: Pull complete                                                                                                                                                                                                                  943879e53568: Pull complete                                                                                                                                                                                                                  e8e263d89203: Downloading [======>                                            ]   8.65MB/70.72MB                                                                                                                                             35cee808fda4: Download complete                                                                                                                                                                                                              44b9fda5ee68: Download complete                                                                                                                                                                                                              399957cb4c77: Download complete                                                                                                                                                                                                              f209389c505c: Download complete                                                                                                                                                                                                              4e6d133e6986: Downloading [==========>                                        ]  2.063MB/9.553MB                                                                                                                                             69477f6b3bb0: Downloading [>                                                  ]  539.6kB/83.67MB                                                                                                                                             76e354953011: Waiting                                                                                                                                                                                                                        148650535be5: Waiting                                                                                                                                                                                                                        5fadf3626025: Waiting                                                                                                                                                                                                                        62fd29264e17: Waiting                                                                                                                                                                                                                        4cb39a3816b8: Waiting                                                                                                                                                                                                                        9fa100f901f9: Pulling fs layer                                                                                                                                                                                                               426cec8046b2: Waiting                                                                                                                                                                                                                        f4586a30b310: Waiting                                                                                                                                                                                                                        d0783e8b3eed: Waiting                                                                                                                                                                                                                        7470218f747a: Waiting                                                                                                                                                                                                                        5acd4467f2ee: Waiting                                                                                                                                                                                                                        ba126c55d9d6: Waiting                                                                                                                                                                                                                        e4a60ff913df: Waiting                                                                                                                                                                                                                        32a43ffe533a: Waiting                                                                                                                                                                                                                        b5dd05fe8354: Waiting                                                                                                                                                                                                                        79fdfd236156: Waiting                                                                                                                                                                                                                        a93bfd51a0dc: Waiting                                                                                                                                                                                                                        0de61090adb9: Waiting                                                                                                                                                                                                                        9ca2493c826d: Waiting                                                                                                                                                                                                                        b99167d55017: Waiting                                                                                                                                                                                                                        f8655de7e9a3: Waiting                                                                                                                                                                                                                        0d2e30c46b4e: Waiting                                                                                                                                                                                                                        cf9038341305: Waiting                                                                                                                                                                                                                        4ba1754de392: Waiting                                                                                                                                                                                                                        409af6a385b5: Waiting                                                                                                                                                                                                                        8c4de03a0c83: Waiting                                                                                                                                                                                                                        66e72de1e224: Waiting                                                                                                                                                                                                                        e287175d9915: Waiting                                                                                                                                                                                                                        89732bc75041: Waiting                                                                                                                                                                                                                        edb569eda09b: Waiting                                                                                                                                                                                                                        3f285ef6f715: Waiting                                                                                                                                                                                                                        1d05c510e148: Waiting                                                                                                                                                                                                                        ca8a463e3eb4: Pulling fs layer                                                                                                                                                                                                               3c2d25d613d7: Pulling fs layer                                                                                                                                                                                                               59972f2e2c6d: Waiting

**Comment by chandan-vibrenthealth** [comment]
I am also facing the same issue with jboss/keycloak

**Comment by matthogan** [comment]
> THE ISSUE IS RESOLVED.
> THE MAIN PROBLEM WAS IN THE TIMEOUT ERROR. SO JUST TO RESOLVE THIS CHANGE THE WIFI CONNECTION WHENEVER U GOT STUCK AND THEN CHANGE IT BACK.WORKS EVERYTIME.KUDOS!!!!!!!!!!!!!!

For whatever reason this worked in my case.

**Comment by jypang86** [comment]
Here to leave my comment in 2021.... Observing the same issue while pulling docker-bench-security.
Tried the same operation on another machine and it was working well.

Tried restart docker even rebooted the machine it wont help......

**Comment by RafaelHGBotelho** [comment]
Same problem here, the images freeze and the process goes extremely slowly. I executed the command on two different machines and by Azure DevOps, but it is stuck until the moment that decides to flow.

**Comment by victor-ramos** [comment]
I was facing the same issue on macOS Catalina 10.17.7 with Docker Desktop 3.2.2.
Restarting the docker application and rebooting the computer did not work for me.

I had to disable my Anti-Malware Symantec Endpoint Protection be able to pull an image.

**Comment by Nahemah1022** [comment]
Have the same issue, and I solved it by restarting docker engine.

**Comment by btraas** [comment]
Switching from Ethernet to a WiFi connection worked for me (after restarting the docker service).

**Comment by leimao** [comment]
Is `systemctl restart docker` a permanent solution?

**Comment by leimao** [comment]
Is `systemctl restart docker` a permanent solution?

**Comment by Luca-Terrazzan** [comment]
Why is this closed? It's clearly not resolved

**Comment by mattyf** [comment]
Encountered on Synology Docker as well.

**Comment by pengfeinie** [comment]
Same issue here, if you guys find way to fix the issue , please tell me, thanks.

**Comment by fcarta29** [comment]
Samesies - super annoying as well as the Docker upgrade/restart pop up

**Comment by sanak** [comment]
Well, I encountered the same issue on macOS Catalina environment, and after stopping Kaspersky Internet Security v21.1, I could pull the image.
So, I guess that some virus software's settings may be one of the cause of this issue.

**Comment by egor-ryashin** [comment]
I wonder if it can be an internet provider and what should be the steps to debug the pulling process?

**Comment by shrinktofit** [comment]
Bump

**Comment by egor-ryashin** [comment]
At least an idea how to debug that would be helpful indeed.

**Comment by egor-ryashin** [comment]
It was antivirus Kaspersky. Switching it off didn't help as it stayed as background process and continued to block, I terminated the process `kavd` manually.

How I found it (in case you have another antivirus):
1. Find docker vpnkit process PID `ps axu | grep docker.vpn`
2. Find what destination port it connects to: `lsof -Pnl +M -i4 | grep <PID>`
```
com.docke 11138      501   69u  IPv4 0x217c40f28f      0t0  TCP 127.0.0.1:52603->127.0.0.1:1110 (FIN_WAIT_2)
com.docke 11138      501   70u  IPv4 0x21737be28f      0t0  TCP 127.0.0.1:52605->127.0.0.1:1110 (FIN_WAIT_2)
com.docke 11138      501   71u  IPv4 0x2172028f      0t0  TCP 127.0.0.1:52607->127.0.0.1:1110 (FIN_WAIT_2)
com.docke 11138      501   72u  IPv4 0x2172ab58af      0t0  TCP 127.0.0.1:52609->127.0.0.1:1110 (FIN_WAIT_2)
com.docke 11138      501   73u  IPv4 0x21e7ecf      0t0  TCP 127.0.0.1:52612->127.0.0.1:1110 (FIN_WAIT_2)
com.docke 11138      501   74u  IPv4 0x217295c12f      0t0  TCP 127.0.0.1:52614->127.0.0.1:1110 (FIN_WAIT_2)
com.docke 11138      501   75u  IPv4 0x21723a628f      0t0  TCP 127.0.0.1:52616->127.0.0.1:1110 (FIN_WAIT_2)
```
3. Find the port process: `sudo lsof -i -P | grep LISTEN | grep 1110`
```
kavd        145           root  148u  IPv4 0x2179f81ecf      0t0    TCP *:1110 (LISTEN)
```

**Comment by btraas** [comment]
@egor-ryashin I think ctrl-t will also give some insight whether the download stream(s) have stalled

**Comment by dzmitry-kankalovich** [comment]
Had the same problem with the `postgres` image recently. Not sure of the root cause, but limiting concurrent connections to 1 helped:
```json
{
  "max-concurrent-downloads": 1 
}
```

**Comment by ingomueller-net** [comment]
Very curious... I am experiencing the same problem using Amazon Linux in EC2 (and a different image). Interestingly, the download progress is not zero; it is just *very* slow. The most curious thing is that, after trying with a different instance, the download was fast until the point that the first instance has reached and become slow after that. It looks like some server in the docker hub (?) or a cache layer in between (?) is slowing down the download of this specific layer...

**Comment by SmetDenis** [comment]
I have the same issue :(
Any thoughts on this moment?

**Comment by Reeebuuk** [comment]
Had same issue. Restarting docker machine didn't help but restarting whole mac resolved the issue for some reason :/
Setting max-concurrent-downloads didn't help.

**Comment by Niek** [comment]
Same issue here, it seems to happen on specific layers. Changing `max-concurrent-downloads` didn't help for me.

**Comment by daylioti** [comment]
Same issue, windows WSL2 for bitnami/solr:8

**Comment by harveydobson** [comment]
So I've had this issue for the past few weeks and just today realised the cause was switching to Kaspersky Antivirus. If I pause the protection, there is no interruption to my downloads. 

So I'm happy - I just wish I had found that out before destroying my entire docker environment... which by the way if you were looking to do, this is a very good guide for mac https://www.mytecbits.com/apple/macos/clean-way-to-uninstall-docker

**Comment by chris-metz** [comment]
Having the same issue on 2 local machines (Windows w/ WSL2 and macos respectively). On both machines when pulling images it starts quiet fast and then slows down to an almost zero speed.

However, pulling images on a VPS works fast without any problems.

**Comment by salesfelipe** [comment]
I was having this same issue on my local machine (MAC OSX). It was solved when I closed the Kapersky application and restarted the docker app as explained above.

**Comment by egor-ryashin** [comment]
I think at this point Kaspersky ought to fix its bugs, so many people turn it off to be able to work.

**Comment by imithu** [comment]
# 19-Feb-2022
still same issue.

**Comment by oharkins** [comment]
### March 16, 2022 
Same Issue Still Happening

**Comment by johnib** [comment]
For me it happened on streams that were interrupted when I cancelled the pull with `ctrl+c` and then executed it again.
Not sure what can be done to recover from this state.. it seems like consequtive pulls do not start downloading from scratch, but from where the stream was intrrupted.

**Comment by oharkins** [comment]
Based on the fact that mine just eventually finished I think the process would need to be killed I just looked later and they had finished even tho I ctrl+c out of the process.

**Comment by Emememssss** [comment]
okay thanks. I also resolved it by halting the process, then restarting the
laptop/ or would be the vm in other cases.

**Comment by mervanerdem** [comment]
`systemctl restart docker ` in ubuntu worked for me

**Comment by JuanFontes** [comment]
Restarting docker worked for me. 
<img width="254" alt="image" src="https://user-images.githubusercontent.com/8904414/163888910-8d568f1a-57bb-411e-8bbd-990b4275598e.png">

**Comment by ibrt** [comment]
Issue is real and it's quite bizarre that it still exists 4 years after being first reported. One thing I noticed is that it might be network-dependent, i.e. it *never* happened to me until I started working at a coworking space which uses https://www.nuvisions.net as provider. Connection is actually pretty solid, never experienced other problems. But this issue now happens to me consistently, only on this network.

**Comment by akhilesh84** [comment]
> Restarting docker worked for me.
> <img alt="image" width="254" src="https://user-images.githubusercontent.com/8904414/163888910-8d568f1a-57bb-411e-8bbd-990b4275598e.png">

For me too...

**Comment by NedislavKolev** [comment]
I had the same problem with Docker on Windows with WSL restarting it fixed the issue.

**Comment by wuduanq** [comment]
same issue in ubuntu vm, for 6 years

**Comment by wuduanq** [comment]
it's time to close the issue

**Comment by slavanorm** [comment]
its funny that i have "stuck" progress visually, but then it turned out that it just turns to "Download Complete" without updating the progress numbers

**Comment by HeikoNardmann** [comment]
I have the same issue inside a Win11 VirtualBox VM. Even with "max-concurrent-downloads" set to 1 it does not help. I get the error message "unknown blob". And the layers downloaded last time are re-downloaded once again ... which is annoying even more.

**Comment by najb121** [comment]
I was facing the same issue but after hitting ctrl-C the download seemed to resume

**Comment by eben0** [comment]
Pruning all build cache worked for me:

`docker builder prune -a`

**Comment by nicks** [comment]
Hi all - I'm going to close this issue. The original reporter said that the problem was fixed for them. Since then, the issue has become a dumping ground for random unrelated pull problems -- possibly caused by faulty networks, faulty anti-virus software, or misunderstandings about registry auth.

If you believe there's a bug that persists for you, please file a new issue with detailed repro steps and/or diagnostics. This allows us to track and fix problems more effectively. Thanks!
